dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -377,6 +388,31 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
         return url;
     }
 
+    static DockerSpecifier determineImageSpecifier(String image, DockerImageReference imageReference) {
+        DockerSpecifier dockerSpecifier = null;
+        String latestTag = "latest";
+        String digestSpecifer = "@sha256:";
```

Comment:
Does the `@` make it impossible to have a tag masquerade as a digest? I hope so; that will simplify things. If not, we may need to do additional validation.

Reply:
A tag can't have the `@` symbol so it's impossible to have a tag masquerade as a digest ([source](https://docs.docker.com/engine/reference/commandline/tag/#extended-description)). However, if the user decided to include a fake image with a tag that looks like `:faketag@sha256:<digest>` in the workflow, then when the workflow gets parsed, the image would incorrectly be classified as a digest specifier. This might not be a big concern because the image won't get saved either way since the image doesn't exist.